crates/storage/codecs/src/alloy/withdrawal.rs

```diff
@@ -0,0 +1,172 @@
+use crate::Compact;
+use alloy_eips::eip4895::Withdrawal;
+use alloy_primitives::Address;
+use bytes::Buf;
+use modular_bitfield::{
+    error,
+    prelude::{Specifier, B4},
+    private::write_specifier,
+};
+
+/// Implement `Compact` for `Withdrawal`.
+impl Compact for Withdrawal {
+    fn to_compact<B>(self, buf: &mut B) -> usize
+    where
+        B: bytes::BufMut + AsMut<[u8]>,
+    {
+        let mut flags = WithdrawalFlags::default();
+        let mut buffer = bytes::BytesMut::new();
+        let index_len = self.index.to_compact(&mut buffer);
+        flags.set_index_len(index_len as u8);
+        let validator_index_len = self.validator_index.to_compact(&mut buffer);
+        flags.set_validator_index_len(validator_index_len as u8);
+        let _address_len = self.address.to_compact(&mut buffer);
+        let amount_len = self.amount.to_compact(&mut buffer);
+        flags.set_amount_len(amount_len as u8);
+        let flags = flags.into_bytes();
+        let total_length = flags.len() + buffer.len();
+        buf.put_slice(&flags);
+        buf.put(buffer);
+        total_length
+    }
+
+    fn from_compact(buf: &[u8], _: usize) -> (Self, &[u8]) {
+        let (flags, mut buf) = WithdrawalFlags::from(buf);
+        let (index, new_buf) = u64::from_compact(buf, flags.index_len() as usize);
+        buf = new_buf;
+        let (validator_index, new_buf) =
+            u64::from_compact(buf, flags.validator_index_len() as usize);
+        buf = new_buf;
+        let (address, new_buf) = Address::from_compact(buf, buf.len());
+        buf = new_buf;
+        let (amount, new_buf) = u64::from_compact(buf, flags.amount_len() as usize);
+        buf = new_buf;
+        let obj = Withdrawal { index, validator_index, address, amount };
+        (obj, buf)
+    }
+}
+
+/// Flags to store metadata about compacted Withdrawal.
+pub(crate) struct WithdrawalFlags {
+    bytes: [u8; 2], // Calculated from B4 Specifier
+}
```

Comment:
```
use modular_bitfield::prelude::*;
                #[doc = "Fieldset that facilitates compacting the parent type. Used bytes: 2 | Unused bits: 1"]
                #[bitfield]
                #[derive(Clone, Copy, Debug, Default)]
                pub struct TestStructFlags {
                    pub f_u64_len: B4,
                    pub f_u256_len: B6,
                    pub f_bool_t_len: B1,
                    pub f_bool_f_len: B1,
                    pub f_option_none_len: B1,
                    pub f_option_some_len: B1,
                    pub f_option_some_u64_len: B1,
                    #[skip]
                    unused: B1,
                }
```

If you implement it as above (taken from one of our tests), all the impl below will be auto generated